ibis/backends/dask/execution/generic.py

```diff
@@ -339,7 +342,11 @@ def execute_cast_series_date(op, data, type, **kwargs):
 @execute_node.register(ops.Limit, dd.DataFrame, integer_types, integer_types)
 def execute_limit_frame(op, data, nrows, offset, **kwargs):
     # NOTE: Dask Dataframes do not support iloc row based indexing
-    return data.loc[offset : (offset + nrows) - 1]
```

Comment:
This didn't behave correctly with multi-partition dataframes.
If the index of `data` wasn't consecutive this also wouldn't select the correct number of rows.